server/src/main/java/org/elasticsearch/index/mapper/FieldAliasMapper.java

```diff
@@ -124,6 +125,11 @@ public Mapper.Builder parse(String name, Map<String, Object> node, MappingParser
             }
             return builder.path(path);
         }
+
+        @Override
+        public boolean supportsVersion(Version indexCreatedVersion) {
+            return true;
```

Comment:
this I find a bit surprising especially given that field aliases were added with 6.4 . I wonder if this is never called, in which case maybe we should throw exception instead, or if it could rely on the default impl instead.

Reply:
Even though field aliases were only added in ES 6.4, they could be used with older indices as well (i.e. you had a 5.x index for example that you kept using in 6.4 and now added the field alias to the mapping of that 5.x index). As field aliases are more of a runtime property (e.g. just like runtime fields, which you could also use on indices created before the release of runtime fields) and have no connection to the actual data, I see no reason to limit their use.